README.md

```diff
@@ -1 +1,187 @@
+        expect(MathService.sum([1,23]).toEqual(24);
+    });
+});
+```
+
+Yes, it's that simple!
+
+## Testing $http calls
+
+Now, our services are also going to be calling `$http` a lot, and it would be a lot of effort to mock an entire backend. Luckily, ngMocks provides us with a nice little tool called `$httpBackend`. This allows us to mock our API calls.
+
+There are three parts to `$httpBackend`:
+
+### $httpBackend.when
+
+We use `$httpBackend` to setup the responses to our HTTP calls. It accepts a method and a URL, and allows us to define the response that we'd get back.
+
+If we have a backend endpoint at `/rest/user` that responds with the current user's information, we can mock it as follows:
+
+```js
+describe('UserService', function () {
+    var $controller;
+
+    beforeEach(module('app'));
+
+    var UserService;
+
+    beforeEach(inject(function ($injector) {
+        UserService = $injector.get('UserService');
+
+        $httpBackend.when('GET', '/rest/user').respond({user: 'Bill Gates', email: 'bill@microsoft.com'});
+    }));
+});
+```
+
+### $httpBackend.expectGET
+
+Now, when we actually test the service's function that calls that endpoint, we need to tell ngMocks that we're expecting the request to occur. We do this by calling `$httpBackend.expectGET` with the endpoint we're expecting to have a request to.
+
+```js
+describe('UserService', function () {
+    var $controller;
+
+    beforeEach(module('app'));
+
+    var UserService;
+
+    beforeEach(inject(function ($injector) {
+        UserService = $injector.get('UserService');
+
+        $httpBackend.when('GET', '/rest/user').respond({user: 'Bill Gates', email: 'bill@microsoft.com'});
+    }));
+
+    it('should get the current users information', function (done) {
+        $httpBackend.expectGET('/rest/user');
+    });
+});
+```
+
+We're now setup to receive the mocked backend response. In this case, we'll get back an object with `name` equal to `Bill Gates` and his email too.
+
+```js
+describe('UserService', function () {
+    var $controller;
+
+    beforeEach(module('app'));
+
+    var UserService;
+
+    beforeEach(inject(function ($injector) {
+        UserService = $injector.get('UserService');
+
+        $httpBackend.when('GET', '/rest/user').respond({user: 'Bill Gates', email: 'bill@microsoft.com'});
+    }));
+
+    it('should get the current users information', function (done) {
+        $httpBackend.expectGET('/rest/user');
+
+        UserService
+          .getUserInfo()
+          .then(function (res) {
+            if (res.email === 'bill@microsoft.com' && res.user === 'Bill Gates') {
+              done();
+            }
+          });
+	});
+});
+```
+
+Here, instead of using `expect().toBe()`, we call a function named `done()` if our response is correct. This means that we can do asynchronous tests in Jasmine.
+
+### $httpBackend.flush()
+
+We then need to call `$httpBackend.flush()` to immediately execute all pending requests.
```

Comment:
I'm confused about what the `flush` method is for - this executes the requests? When do the expectations actually get checked?